tests/mocks/createAlgoliaSearchClient.ts

```diff
@@ -17,19 +21,42 @@ import {
 } from './createAPIResponse';
 
 import type { HostOptions } from '@algolia/transporter';
+import type { SearchClient } from 'algoliasearch-helper/types/algoliasearch';
 
 type OverrideKeys<TTarget, TOptions> = TOptions extends Record<string, never>
   ? TTarget
   : Omit<TTarget, keyof TOptions> & TOptions;
 
-type SearchClient = ReturnType<typeof algoliasearch>;
+type IfAny<TTest, TFallback> = (0 extends 1 & TTest ? true : false) extends true
+  ? IfAny<TFallback, never>
+  : TTest;
+
+function ifActualFunction<TTest, TFallback>(
+  test: TTest,
+  fallback: TFallback
+): IfAny<TTest, TFallback> {
+  if (typeof test === 'function') {
+    return test as IfAny<TTest, TFallback>;
+  }
+  if (typeof fallback === 'function') {
+    return fallback as IfAny<TTest, TFallback>;
+  }
+  throw new Error('Neither test nor fallback is a function');
+}
+
+const algoliasearch = ifActualFunction(
+  namedConstructor,
+  defaultConstructor
+) as unknown as (appId: string, apiKey: string) => SearchClient;
```

Comment:
😳

Reply:
actually this isn't needed anymore, that was when I was still attempting to use the function from the import (but it isn't possible)